include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1051 @@
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumMomPos2,
+                             gSumMomPos2->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedCount_e,
+                             gCount_e->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+
+            /** all non-reduce-root processes are done now */
+            if( !this->isPlaneReduceRoot )
+                return;
+
+            // gather to file writer
+            container::HostBuffer< float_64, DIM1 > globalSumMom2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalSumPos2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalSumMomPos2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalCount_e( subGrid.getGlobalDomain().size.y() );
+
+            // gather y offsets, so we can store our gathered data in the right order
+            int gatherSize = -1;
+            int gatherRank = -1;
+            MPI_CHECK(MPI_Comm_rank( commGather, &gatherRank ));
+            MPI_CHECK(MPI_Comm_size( commGather, &gatherSize ));
+            std::vector< int > y_offsets( gatherSize );
+            std::vector< int > y_sizes( gatherSize );
+            std::vector< int > gatherRanks( gatherSize );
+            long int const y_off = subGrid.getLocalDomain().offset.y();
+            int const y_siz = subGrid.getLocalDomain().size.y();
+
+            MPI_CHECK(MPI_Gather(
+                &y_off,
+                1,
+                MPI_INT,
+                y_offsets.data(),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ));
+            MPI_CHECK(MPI_Gather(
+                &y_siz,
+                1,
+                MPI_INT,
+                y_sizes.data(),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ));
+            MPI_CHECK(MPI_Gather(
+                &gatherRank,
+                1,
+                MPI_INT,
+                gatherRanks.data(),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ));
+
+            auto y_offsets2 = y_offsets;
+            // permute ranks for ascending y offset order
+            // example:
+            //   gatherRank = {  2,  3,  4,  0,  1 }
+            //   y_off      = { 11, 15, 17, 19,  9 }
+            sort_by_key( y_offsets, gatherRanks );
```

Comment:
must be sorted always by `gatherRanks` as key, not by `y_offsets`.
reason: `MPI_Gatherv` gets the data in order of ranks and needs to *place* this data by offset :)